docs/extensibility/entity-extension.md

```diff
@@ -38,41 +38,41 @@ Shopsys\FrameworkBundle\Model\Product\ProductDataFactoryInterface:
 
 The other way of data - from `OrderItem` to `OrderItemData` is standard.
-So if you extend `OrderItem` in a standard fashion, set a new field in extended `OrderItemDataFactory` then `OrderItemData` object will contain the correct value from the `OrderItem` object.
+So, if you extend `OrderItem` in a standard fashion, set a new field in the extended `OrderItemDataFactory`, then the `OrderItemData` object will contain the correct value from the `OrderItem` object.
 
 Creating a new type of `OrderItem` is possible and does not cause problems because the new type is completely in your hands.
 
 You can read about alternative solutions we considered and the reasons behind this approach in [Entity Extension vs. Entity Generation](entity-extension-vs-entity-generation.md).
 
 ## Introduction into entity extensibility
-Let's suppose that we are implementing Dream project as a clone of the [project-base repository](https://github.com/shopsys/project-base).
-Dream project is dependent on the glass-box [framework repository](https://github.com/shopsys/framework).
+Let's suppose that we are implementing the Dream project as a clone of the [project-base repository](https://github.com/shopsys/project-base).
+Dream project depends on the glass-box [framework repository](https://github.com/shopsys/framework).
 The framework is, of course, independent of our Dream project.
 
-Entities in the framework are full-featured themselves and we want to extend them with our custom properties and associations.
+Entities in the framework are full-featured themselves, and we want to extend them with our custom properties and associations.
 We achieve this via inheritance:
 
 ![Dream project entities extend Framework entities](img/entity-extension.png)
 
-Dream project in the example above has extended entities Category and Product.
-An association with a custom entity Dream Product Speciality was added to Product.
-CategoryDomain was not extended, this means that the extended Dream Category has an association with the original CategoryDomain.
+The dream project in the example above has extended entities Category and Product.
+An association with a custom entity, Dream Product Speciality was added to Product.
+CategoryDomain was not extended. This means that the extended Dream Category is associated with the original CategoryDomain.
 
-Doctrine allows us to have only the original entity or the extended entity in the whole system, both are not possible.
-So all associations, new objects, repositories, query builders - everything has to be consistent, otherwise, Doctrine will fail.
+Doctrine allows us to have only the original entity or the extended entity in the whole system. Both are not possible.
+So, everything has to be consistent for all associations, new objects, repositories, query builders. Otherwise, Doctrine will fail.
```

Comment:
```suggestion
So, everything has to be consistent for all associations, new objects, repositories, and query builders otherwise, Doctrine will fail.
```